File: python/leetcode/rotate-array.py
class Solution:
    def rotate(self, nums, k):
        new_arr = [None] * len(nums)
        for i in range(len(nums)):
            new_arr[(i + k) % len(nums)] = nums[i]
        for i in range(len(nums)):
            nums[i] = new_arr[i]

File: java/leetcode/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int l = nums.length;
        int[] newArr = new int[l];
        for (int i = 0; i < l; i++) {
            newArr[(i + k) % l] = nums[i];
        }
        for (int i = 0; i < l; i++) {
            nums[i] = newArr[i];
        }
    }
}
